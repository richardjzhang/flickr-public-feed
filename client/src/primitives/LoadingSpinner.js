// @flow strict
/** @jsx jsx */
import { jsx } from '@emotion/core';
import { colors } from 'src/styles';

export default function LoadingSpinner() {
  return (
    <svg
      css={{
        margin: 'auto',
        background: 'none',
        display: 'block',
        shapeRendering: 'auto',
      }}
      width="150px"
      height="150px"
      viewBox="0 0 100 100"
      preserveAspectRatio="xMidYMid"
    >
      <circle cx="27" cy="50" fill={colors.scienceBlue} r="23">
        <animate
          attributeName="cx"
          repeatCount="indefinite"
          dur="1s"
          keyTimes="0;0.5;1"
          values="27;73;27"
          begin="-0.5s"
        ></animate>
      </circle>
      <circle cx="73" cy="50" fill={colors.rose} r="23">
        <animate
          attributeName="cx"
          repeatCount="indefinite"
          dur="1s"
          keyTimes="0;0.5;1"
          values="27;73;27"
          begin="0s"
        ></animate>
      </circle>
      <circle cx="27" cy="50" fill={colors.scienceBlue} r="23">
        <animate
          attributeName="cx"
          repeatCount="indefinite"
          dur="1s"
          keyTimes="0;0.5;1"
          values="27;73;27"
          begin="-0.5s"
        ></animate>
        <animate
          attributeName="fill-opacity"
          values="0;0;1;1"
          calcMode="discrete"
          keyTimes="0;0.499;0.5;1"
          dur="1s"
          repeatCount="indefinite"
        ></animate>
      </circle>
    </svg>
  );
}
