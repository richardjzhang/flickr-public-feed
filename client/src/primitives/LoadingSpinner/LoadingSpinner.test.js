import { render } from '@testing-library/react';
import * as React from 'react';

import LoadingSpinner from '.';

test('matches snapshot', () => {
  const tree = render(<LoadingSpinner />);
  expect(tree).toMatchSnapshot();
});
