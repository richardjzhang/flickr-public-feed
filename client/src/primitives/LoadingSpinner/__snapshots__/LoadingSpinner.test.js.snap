// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        class="css-1mn86yq-LoadingSpinner"
        height="150px"
        preserveAspectRatio="xMidYMid"
        viewBox="0 0 100 100"
        width="150px"
      >
        <circle
          cx="27"
          cy="50"
          fill="#0063DB"
          r="23"
        >
          <animate
            attributeName="cx"
            begin="-0.5s"
            dur="1s"
            keyTimes="0;0.5;1"
            repeatCount="indefinite"
            values="27;73;27"
          />
        </circle>
        <circle
          cx="73"
          cy="50"
          fill="#FF0084"
          r="23"
        >
          <animate
            attributeName="cx"
            begin="0s"
            dur="1s"
            keyTimes="0;0.5;1"
            repeatCount="indefinite"
            values="27;73;27"
          />
        </circle>
        <circle
          cx="27"
          cy="50"
          fill="#0063DB"
          r="23"
        >
          <animate
            attributeName="cx"
            begin="-0.5s"
            dur="1s"
            keyTimes="0;0.5;1"
            repeatCount="indefinite"
            values="27;73;27"
          />
          <animate
            attributeName="fill-opacity"
            calcMode="discrete"
            dur="1s"
            keyTimes="0;0.499;0.5;1"
            repeatCount="indefinite"
            values="0;0;1;1"
          />
        </circle>
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="css-1mn86yq-LoadingSpinner"
      height="150px"
      preserveAspectRatio="xMidYMid"
      viewBox="0 0 100 100"
      width="150px"
    >
      <circle
        cx="27"
        cy="50"
        fill="#0063DB"
        r="23"
      >
        <animate
          attributeName="cx"
          begin="-0.5s"
          dur="1s"
          keyTimes="0;0.5;1"
          repeatCount="indefinite"
          values="27;73;27"
        />
      </circle>
      <circle
        cx="73"
        cy="50"
        fill="#FF0084"
        r="23"
      >
        <animate
          attributeName="cx"
          begin="0s"
          dur="1s"
          keyTimes="0;0.5;1"
          repeatCount="indefinite"
          values="27;73;27"
        />
      </circle>
      <circle
        cx="27"
        cy="50"
        fill="#0063DB"
        r="23"
      >
        <animate
          attributeName="cx"
          begin="-0.5s"
          dur="1s"
          keyTimes="0;0.5;1"
          repeatCount="indefinite"
          values="27;73;27"
        />
        <animate
          attributeName="fill-opacity"
          calcMode="discrete"
          dur="1s"
          keyTimes="0;0.499;0.5;1"
          repeatCount="indefinite"
          values="0;0;1;1"
        />
      </circle>
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
